src/libraries/System.Private.Xml.Linq/tests/misc/LoadSaveAsyncTests.cs

```diff
@@ -218,53 +218,184 @@ public async Task SaveAsync_CallsAsyncOnly_SaveSync_CallsSyncOnly(bool isAsync,
         {
             XDocument document = XDocument.Parse("<root>Test document async save</root>");
             var element = new XElement("Test");
-            using (ForceSyncAsyncStream stream = new ForceSyncAsyncStream(isAsync))
+            
+            if (isAsync)
             {
-                if (isAsync)
+                await using (ForceSyncAsyncStream stream = new ForceSyncAsyncStream(isAsync))
                 {
                     await document.SaveAsync(stream, saveOptions, CancellationToken.None);
                     await element.SaveAsync(stream, saveOptions, CancellationToken.None);
                 }
-                else
+            }
+            else
+            {
+                using (ForceSyncAsyncStream stream = new ForceSyncAsyncStream(isAsync))
                 {
                     document.Save(stream);
                     element.Save(stream);
                 }
             }
         }
+
+        [Theory]
+        [MemberData(nameof(RoundtripSyncAsyncOnly_MemberData))]
+        public static async Task RoundtripSyncAsyncOnly(bool async, bool document, LoadOptions loadOptions, SaveOptions saveOptions)
+        {
+            ForceSyncAsyncStream outputStream = new ForceSyncAsyncStream(async);
+            
+            if (async)
+            {
+                await using (Stream inputStream = new ForceSyncAsyncStream(async, FilePathUtil.getStream(GetTestFileName())))
+                {
+                    if (document)
+                    {
+                        XDocument asyncDoc = await XDocument.LoadAsync(inputStream, loadOptions, CancellationToken.None);
+                        await asyncDoc.SaveAsync(outputStream, saveOptions, CancellationToken.None);
+                    }
+                    else
+                    {
+                        XElement asyncElement = await XElement.LoadAsync(inputStream, loadOptions, CancellationToken.None);
+                        await asyncElement.SaveAsync(outputStream, saveOptions, CancellationToken.None);
+                    }
+                }
+            }
+            else
+            {
+                using (Stream inputStream = new ForceSyncAsyncStream(async, FilePathUtil.getStream(GetTestFileName())))
+                {
+                    if (document)
+                    {
+                        XDocument syncDoc = XDocument.Load(inputStream, loadOptions);
+                        syncDoc.Save(outputStream, saveOptions);
+                    }
+                    else
+                    {
+                        XElement syncElement = XElement.Load(inputStream, loadOptions);
+                        syncElement.Save(outputStream, saveOptions);
+                    }
+                }
+            }
```

Comment:
Could we assert/compare sync async output too?

Reply:
Done. But now this test seems very close to the existing one named `RoundtripSyncAsyncMatches_Stream`. Should we remove it?

Reply:
I think we can keep it, as this one specifically for forced sync async stream